build_tools/pkgci/build_linux_packages.sh

```diff
@@ -202,6 +208,12 @@ function build_iree_compiler() {
   build_wheel compiler/
 }
 
+function build_iree_pjrt_plugin() {
+  # the first parameter can be `cpu`, `cuda`, `rocm` or `vulkan`
+  local pjrt_platform=$1
+  build_wheel integrations/pjrt/python_packages/iree_${pjrt_platform}_plugin/
+}
```

Comment:
Thinking about this some more, both `build_linux_packages.sh` scripts are where the pjrt plugins should likely live eventually, but I'd want to think through the package deployment story more before setting things up that way. Building the packages right where they are tested (in a `pkgci_test_jax.yml`/`pkgci_test_pjrt.yml`/`pkgci_test_jax_pjrt.yml`) should be less plumbing and boilerplate while still taking steps in that eventual direction:

1. Continuous builds
2. Continuous tests
3. Continuous release builds to nightly packages
4. Stable release builds to PyPI

I'm not sure how steps 3/4 should look with PJRT / JAX. Would we publish a single `iree-jax` package with extras for each platform, e.g. `iree-jax[cuda]`? One package per platform, e.g. `iree-jax-cuda`? Bundle into the existing packages somehow? Continue to require users to build from source?

In iree-turbine, which interfaces with PyTorch, we implement code to register a custom backend:
* https://github.com/iree-org/iree-turbine/blob/f8e0cbbbf89582f7fbcbb4167a95be717d4fd201/setup.py#L100-L104
* https://pytorch.org/docs/stable/torch.compiler_custom_backends.html
* (though we currently only use that for a limited subset of what we could do eventually, see https://github.com/iree-org/iree-turbine/issues/78)

Maybe JAX / PJRT has or should have a similar mechanism, so installing the package is enough to enable it? Or are the existing APIs like `JAX_PLATFORMS` sufficient for a good packaging / deployment user story?
* https://jax.readthedocs.io/en/latest/about.html#a-pluggable-backend
* https://jax.readthedocs.io/en/latest/faq.html#controlling-data-and-computation-placement-on-devices

Reply:
Reaonable. I'll rewrite these scripts to separate them from `build_linux_packages.sh` for now.

> Maybe JAX / PJRT has or should have a similar mechanism, so installing the package is enough to enable it?

Yeah, installing the `iree_pjrt_plugin_cpu/cuda/..` package will lead to a registration in JAX so that `iree_cpu/cuda/..` will be available to be used in `JAX_PLATFORMS`.

Besides JAX, for TensorFlow it seems we (potentially) can also use the PJRT plugin to inject IREE as a [NextPlugableDevice](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/common_runtime/next_pluggable_device) so that TF users can use IREE without compilcated export steps.